src/group-constructors.js

```diff
@@ -37,6 +41,61 @@
   };
 
   constructor.prototype = {
+    _isAncestor: function(effect) {
+      var a = this;
+      while (a !== null) {
+        if (a == effect)
+          return true;
+        a = a._parent;
+      }
+      return false;
+    },
+    _putChild: function(args, isAppend) {
+      var message = isAppend ? 'Cannot append an ancestor or self' : 'Cannot prepend an ancestor or self';
+      for (var i in args) {
```

Comment:
Ha. Oops. I've done this a couple of times. Both times I went to used for..of and then remembered I can't.